clicommand/pipeline_upload.go

```diff
@@ -613,17 +611,20 @@ func (cfg *PipelineUploadConfig) parseAndInterpolate(ctx context.Context, src st
 
 // gatherChangedFiles determines changed files in this build.
 func gatherChangedFiles(diffBase string) (mergeBase string, changedPaths []string, err error) {
-	// The --merge-base flag was only added to git-diff recently.
+	// Use three-dot syntax to find the merge base and diff in one command.
+	// This handles both feature branches and commits directly on main.
+	gitDiff, err := exec.Command("git", "diff", "--name-only", diffBase+"...HEAD").Output()
```

Comment:
I'm curious to understand how exactly this addresses the problem (where the commit is directly to main). If `diffBase == "main"` and `HEAD` is the head of `main`, then `git diff --name-only main...HEAD` is still nothing, right?